File: core/src/main/java/com/toocol/termio/core/file/handlers/BlockingChooseFileHandler.java
package com.toocol.termio.core.file.handlers;

import com.toocol.termio.core.file.core.FileChooser;
import com.toocol.termio.utilities.module.IAddress;
import com.toocol.termio.utilities.module.BlockingMessageHandler;
import com.toocol.termio.core.file.FileAddress;
import io.vertx.core.AsyncResult;
import io.vertx.core.Context;
import io.vertx.core.Promise;
import io.vertx.core.Vertx;
import io.vertx.core.eventbus.Message;
import org.jetbrains.annotations.NotNull;

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/10 19:34
 * @version: 0.0.1
 */
public final class BlockingChooseFileHandler extends BlockingMessageHandler<String> {

    public BlockingChooseFileHandler(Vertx vertx, Context context, boolean parallel) {
        super(vertx, context, parallel);
    }

    @NotNull
    @Override
    public IAddress consume() {
        return FileAddress.CHOOSE_FILE;
    }

    @Override
    protected <T> void handleBlocking(@NotNull Promise<String> promise, @NotNull Message<T> message) throws Exception {
        FileChooser fileChooser = new FileChooser();
        promise.complete(fileChooser.showOpenDialog());
    }

    @Override
    protected <T> void resultBlocking(@NotNull AsyncResult<String> asyncResult, @NotNull Message<T> message) throws Exception {
        message.reply(asyncResult.result());
    }
}

File: core/src/main/java/com/toocol/termio/core/file/handlers/BlockingChooseFileHandler.kt
package com.toocol.termio.core.file.handlers

import com.toocol.termio.core.file.FileAddress
import com.toocol.termio.core.file.core.FileChooser
import com.toocol.termio.utilities.module.BlockingMessageHandler
import com.toocol.termio.utilities.module.IAddress
import io.vertx.core.AsyncResult
import io.vertx.core.Context
import io.vertx.core.Promise
import io.vertx.core.Vertx
import io.vertx.core.eventbus.Message

/**
 * @author ：JoeZane (joezane.cn@gmail.com)
 * @date: 2022/4/10 19:34
 * @version: 0.0.1
 */
class BlockingChooseFileHandler(vertx: Vertx?, context: Context?, parallel: Boolean) : BlockingMessageHandler<String?>(
    vertx!!, context!!, parallel) {
    override fun consume(): IAddress {
        return FileAddress.CHOOSE_FILE
    }

    @Throws(Exception::class)
    override fun <T> handleBlocking(promise: Promise<String?>, message: Message<T>) {
        val fileChooser = FileChooser()
        promise.complete(fileChooser.showOpenDialog())
    }

    @Throws(Exception::class)
    override fun <T> resultBlocking(asyncResult: AsyncResult<String?>, message: Message<T>) {
        message.reply(asyncResult.result())
    }
}